### PR TITLE
[READY] LIZARD WINGS for the flight potion, flight potion no longer makes you new species

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

WINGS POTION GIVES LIZARDS LIZARD WINGS

image
WOWZERS

## Why It's Good For The Game

IT WAS A FREAKING SHAME THIS DIDN'T HAPPEN BEFORE THIS POINT AND NOW LIZARD MINERS CAN USE THE POTION AS WELL

## Changelog
:cl: Wjohnston sprites and Mey Ha Zah for some help getting those sprites into a sane form
add: the flight potion now works with lizards
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
lizards dont deserve wings BUT HOLY FUCK THOSE WINGFS

**Comment by MMMiracles** [comment]
pros: you can fly
cons: you can't fit through 1-tile hallways anymore

**Comment by tralezab** [comment]
angels get free holiness that dragonborn don't, so i'll leave off any downsides

**Comment by Mey-Ha-Zah** [comment]
Only problem for Angels is wing envy. So... hah.

**Comment by Dawson1917** [comment]
The potion is angelic. It's supposed to grant the wings of an angel, not let you LARP as Spyro the dragon.

Nice sprite though.

**Comment by necromanceranne** [comment]
Why can't humies have dragon wings. I'm jealous as hecc.

**Comment by tralezab** [comment]
@Dawson1917 
It just says flight potion, let people larp as spyro pls and ty,

**Comment by Dawson1917** [comment]
> 
> 
> @Dawson1917
> It just says flight potion, let people larp as spyro pls and ty,

make the potion turn lizards purple then
only way i support this

**Comment by Nickvr628** [comment]
Merge these and unironically give them to vampires

**Comment by TerraGS** [comment]
This is now a scaley server. Non-lizards mad.

**Comment by Ratvar** [comment]
Give this to vampires

**Comment by Tlaltecuhtli** [comment]
does it make the lizard valid

**Comment by nemvar** [comment]
This probably makes more sense as the ashdrake potion result rather than the angel one.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -306,6 +310,15 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 		for(var/i in inherent_factions)
 			C.faction += i //Using +=/-= for this in case you also gain the faction from a different source.
 
+	if(flying_species && isnull(fly))
+		fly = new
+		fly.Grant(C)
+	if(C.dna && C.dna.species && (C.dna.features["wings"] != wings_icon))
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: This doesn't make sense. C.dna.species is src.

**Comment by Mey-Ha-Zah** [comment]
nemvar, flight potion literally just gives flight to one race. Adding more races to it makes sense over locking wings behind a megafauna kill, and adding more rng to an already full pool.

Imagine you were a miner, you just killed your first ash drake. You already have angel wings and you heard about the mystical and rare reward of drake blood. You drink up, hoping to either get lava feet or dragon form. Nope, your angel wings turn into dragon wings. Bummer.

Adding it to flight potion just increases the use of flight potion, rather than pissing off lizard miners who know they can only give it to a human for it to have any use, losing a tendril reward.

**Comment by nemvar** [comment]
I meant that you can already just randomly turn into a lizard when you drink the ashdrake potion. That's really lame and this would be better for that. If you want the flight potion to work for all races, rework wings into an organ. Just adding more shit onto it won't solve that problem.

**Comment by Mey-Ha-Zah** [comment]
Thats a whole other thing, but I get the idea, this is just a simple addition to make flight potion less of a snowflake item by giving it more interactions.

**Comment by dr-bee** [comment]
> 
> 
> If you want the flight potion to work for all races, rework wings into an organ.

I want this now just so I can cut people's wings off as a med doctor.

**Comment by tralezab** [comment]
Nemvar invited me out to brunch to have an intellectual conversation about this pr away from the masses of mouth breathers and safe to say I'm boutta refactor the shit out of this

**Comment by Shadowflame909** [comment]
nemvar has a good idea

Im fine with this but it'd be cool for the drake-blood lizard from the ash drake potion give you

1- some wyvern wings

2- An ancient shout skill

Fus roh dah.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+	else
+		for(var/i=1, i<5, i++)
+			spawn (i)
+				step(H, olddir)
+				H.spin(1,1)
+	return 1
+
+/datum/species/proc/ToggleFlight(mob/living/carbon/human/H,flight)
+	if(flight && CanFly(H))
+		stunmod = 2
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: stunmod *= 2

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+		speedmod = -0.35
+		H.setMovetype(H.movement_type | FLYING)
+		override_float = TRUE
+		H.pass_flags |= PASSTABLE
+		H.OpenWings()
+	else
+		stunmod = 1
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: stunmod *= 0.5

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+		for(var/i=1, i<5, i++)
+			spawn (i)
+				step(H, olddir)
+				H.spin(1,1)
+	return 1
+
+/datum/species/proc/ToggleFlight(mob/living/carbon/human/H,flight)
+	if(flight && CanFly(H))
+		stunmod = 2
+		speedmod = -0.35
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: speedmod -= 0.35

File: code/modules/mob/living/carbon/human/species.dm [hunk 5]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+		H.setMovetype(H.movement_type | FLYING)
+		override_float = TRUE
+		H.pass_flags |= PASSTABLE
+		H.OpenWings()
+	else
+		stunmod = 1
+		speedmod = 0
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: speedmod += 0.35

File: code/modules/mob/living/carbon/human/species.dm [hunk 6]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+/datum/action/innate/flight
+	name = "Toggle Flight"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUN
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "flight"
+
+/datum/action/innate/flight/Activate()
+	var/mob/living/carbon/human/H = owner
+	var/datum/species/angel/A = H.dna.species
+	if(A.CanFly(H))
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: This is already checked in ToggleFlight. There's probably some way to do this without calling this twice.

File: code/modules/mob/living/carbon/human/species.dm [hunk 7]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+		buckled_obj.unbuckle_mob(H)
+		step(buckled_obj, olddir)
+	else
+		for(var/i=1, i<5, i++)
+			spawn (i)
+				step(H, olddir)
+				H.spin(1,1)
+	return 1
+
+/datum/species/proc/ToggleFlight(mob/living/carbon/human/H,flight)
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: The ``flight`` arg doesn't really make sense here. This is a toggle not a setter. At least the name would imply that. This is just asking for stuff to break.

File: code/modules/mob/living/carbon/human/species.dm [hunk 8]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+
+	var/olddir = H.dir
+
+	H.stop_pulling()
+	if(buckled_obj)
+		buckled_obj.unbuckle_mob(H)
+		step(buckled_obj, olddir)
+	else
+		for(var/i=1, i<5, i++)
+			spawn (i)
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: ``new /datum/forced_movement(C, get_ranged_target_turf(C, olddir, 4), 1, FALSE, CALLBACK(C, /mob/living/carbon/.proc/spin, 1, 1))``

File: code/modules/mob/living/carbon/human/species.dm [hunk 9]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+	button_icon_state = "flight"
+
+/datum/action/innate/flight/Activate()
+	var/mob/living/carbon/human/H = owner
+	var/datum/species/angel/A = H.dna.species
+	if(A.CanFly(H))
+		if(H.movement_type & FLYING)
+			to_chat(H, "<span class='notice'>You settle gently back onto the ground...</span>")
+			A.ToggleFlight(H,FALSE)
+			H.update_mobility()
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: This should probaby be called at the end of ToggleFlight

File: code/modules/mob/living/carbon/human/species.dm [hunk 10]
```diff
@@ -1798,3 +1832,108 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+
+/datum/action/innate/flight
+	name = "Toggle Flight"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUN
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "flight"
+
+/datum/action/innate/flight/Activate()
+	var/mob/living/carbon/human/H = owner
+	var/datum/species/angel/A = H.dna.species
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: ``var/datum/species/A = H.dna.species``

File: code/modules/mob/living/carbon/human/species.dm (hunk 9 above)
Comment (tralezab) [reply code/modules/mob/living/carbon/human/species.dm]: nope you're right

File: code/modules/mob/living/carbon/human/species.dm [hunk 11]
```diff
@@ -1798,3 +1832,104 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+
+	var/olddir = H.dir
+
+	H.stop_pulling()
+	if(buckled_obj)
+		buckled_obj.unbuckle_mob(H)
+		step(buckled_obj, olddir)
+	else
+		for(var/i in 1 to 5)
+			new /datum/forced_movement(C, get_ranged_target_turf(C, olddir, 4), 1, FALSE, CALLBACK(C, /mob/living/carbon/.proc/spin, 1, 1))
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: You only need to do this once. Also replace C with H, my bad.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 12]
```diff
@@ -609,13 +609,13 @@
 
 /datum/reagent/flightpotion/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message = 1)
 	if(iscarbon(M) && M.stat != DEAD)
-		if(!ishumanbasic(M) || reac_volume < 5) // implying xenohumans are holy
+		if(!(ishumanbasic(M) || islizard(M)) || reac_volume < 5) // implying xenohumans are holy
```
Comment (nemvar) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Don't forget to remove this. Possibly even keep the holy trait for basic humans.

File: code/modules/mob/living/carbon/human/species.dm [hunk 13]
```diff
@@ -1798,3 +1832,104 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+/datum/species/proc/GiveSpeciesFlight(mob/living/carbon/human/H)
+	if(flying_species) //species that already have flying traits should not work with this proc
+		return
+	flying_species = TRUE
+	if(isnull(fly))
+		fly = new
+		fly.Grant(H)
+	if(H.dna.features["wings"] != wings_icon)
+		if(!("wings" in H.dna.species.mutant_bodyparts))
+			H.dna.species.mutant_bodyparts |= "wings"
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]: The ``"wings" in mutantparts`` check isn't needed. Just do ``mutant_bodyparts |= "wings"`` without the check or use += if you want to keep it.

Also, ``H.dna.species = src``

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 14]
```diff
@@ -609,13 +609,13 @@
...
 			if(method == INGEST && show_message)
 				to_chat(M, "<span class='notice'><i>You feel nothing but a terrible aftertaste.</i></span>")
 			return ..()
 
 		to_chat(M, "<span class='userdanger'>A terrible pain travels down your back as wings burst out!</span>")
-		M.set_species(/datum/species/angel)
+		GiveSpeciesFlight(M)
```
Comment (nemvar) [inline code/modules/mining/lavaland/necropolis_chests.dm]: This proc is on species and not global

**Comment by DONKARMA** [comment]
It's a holy potion. Non-humans go to hell. Give it to ash drake drops.

**Comment by carlarctg** [comment]
> It's a holy potion. Non-humans go to hell. Give it to ash drake drops.

if you're going to repeat a point someone else already made without any difference whatsoever, refute the counter-arguments given by other people

**Comment by ATH1909** [comment]
What happens if a moth chugs a flight potion now?

**Comment by shellspeed1** [comment]
> 
> 
> What happens if a moth chugs a flight potion now?

Gonna assume nothing, moths need quite a bit work still tbh. Lots of shitcode behind them.

**Comment by tralezab** [comment]
Fuck moths, gonna replace them with sentient calendars called months

File: code/modules/mob/living/carbon/human/species.dm (hunk 11 above)
Comment (tralezab) [reply code/modules/mob/living/carbon/human/species.dm]: 🙏

File: code/modules/mob/living/carbon/human/species.dm (hunk 13 above)
Comment (tralezab) [reply code/modules/mob/living/carbon/human/species.dm]: riight. only adds when it's not there. gotta love |=.

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/mob/living/carbon/human/species.dm]: honestly, if it's a species initial thing, then the wings should just be defined and not set, ya

**Comment by wesoda25** [comment]
tralezab is my hero

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 15]
```diff
@@ -609,16 +609,21 @@
...
-		M.set_species(/datum/species/angel)
-		playsound(M.loc, 'sound/items/poster_ripped.ogg', 50, 1, -1)
-		M.adjustBruteLoss(20)
-		M.emote("scream")
+		to_chat(C, "<span class='userdanger'>A terrible pain travels down your back as wings burst out!</span>")
+		C.dna.species.GiveSpeciesFlight(C)
+		if(holycheck)
+			to_chat(C, "<span class='notice'>You feel blessed!</span>")
+			ADD_TRAIT(C, TRAIT_HOLY, SPECIES_TRAIT)
+		playsound(C.loc, 'sound/items/poster_ripped.ogg', 50, 1, -1)
```
Comment (actioninja) [inline code/modules/mining/lavaland/necropolis_chests.dm]: fourth parameter is a boolean, 1 to TRUE

File: code/modules/mob/living/carbon/human/species.dm [hunk 16]
```diff
@@ -1798,3 +1827,102 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+		return TRUE
+
+/datum/species/proc/flyslip(mob/living/carbon/human/H)
+	var/obj/buckled_obj
+	if(H.buckled)
+		buckled_obj = H.buckled
+
+	to_chat(H, "<span class='notice'>Your wings spazz out and launch you!</span>")
+
+	playsound(H.loc, 'sound/misc/slip.ogg', 50, 1, -3)
```
Comment (actioninja) [inline code/modules/mob/living/carbon/human/species.dm]: same here

**Review by actioninja — CHANGES_REQUESTED** [review]
looks like nemvar already got most of the actual code review out of the way. Small stuff

**Review by actioninja — APPROVED** [review]

**Comment by actioninja** [comment]
how soon until people complain about this

**Comment by carshalash** [comment]
Fuck you

**Comment by adamkad1** [comment]
fus ro dah when

**Pull request merged**